pkg/operator/ceph/cluster/rbd/controller_test.go

```diff
@@ -192,6 +193,7 @@ func TestCephRBDMirrorController(t *testing.T) {
 			context:          c,
 			peers:            make(map[string]*peerSpec),
 			opManagerContext: ctx,
+			recorder:         record.NewFakeRecorder(5),
 		}
 
 		rbdMirror.Spec.Peers.SecretNames = []string{peerSecretName}
```

Comment:
I can't comment on lines below, so I'll comment here. In order to make this test pass, `assert.Error(t, err)` after `r.Reconcile(ctx, req)` must be changed to `assert.NoError()`. If the result `res` is nonzero, then `ReportReconcileResult` will make the error nil when it returns. This is because returning an error will cause controller-runtime to **immediately** requeue the reconcile, but we assume a nonzero value means that the reconcile should be requeued after some delay. Only by returning a nonzero result with a nil error can we do that. 

However... To be honest, it looks like the RBD reconciler `reconcile()` should probably be updated instead. We only want to return `opcontroller.ImmediateRetryResult` if there is no error. If there is an error, we can just return an empty `reconcile.Result{}` since it will be re-reconciled immediately.

Reply:
got it now, thanks @BlaineEXE for explaining all. But, currently, for this pr, I'm not changing the`ImmediateRetryResult` to `reconcile.Result{}` I can do that in follow-up. But I agree that doing so will be the right fix.